crates/revm/src/evm.rs

```diff
@@ -274,19 +274,33 @@ impl<EXT, DB: Database> Evm<'_, EXT, DB> {
 
         shared_memory.new_context();
 
-        // peek last stack frame.
+        // Peek the last stack frame.
         let mut stack_frame = call_stack.last_mut().unwrap();
 
         loop {
-            // run interpreter
-            let interpreter = &mut stack_frame.frame_data_mut().interpreter;
-            let next_action = interpreter.run(shared_memory, instruction_table, self);
+            // Execute the frame.
+            let next_action;
+            'blk: {
+                // Run the execute frame hook if set.
+                if let Some(execute_frame) = self.handler.execution.execute_frame.take() {
+                    let maybe_next_action = execute_frame(stack_frame, &mut shared_memory, self);
+                    self.handler.execution.execute_frame = Some(execute_frame);
+                    if let Some(a) = maybe_next_action {
+                        next_action = a;
+                        break 'blk;
+                    }
+                }
+
+                // Otherwise, or if the hook returned `None`, run the interpreter.
+                let interpreter = stack_frame.interpreter_mut();
+                next_action = interpreter.run(shared_memory, instruction_table, self);
+                // Take the shared memory back.
+                shared_memory = interpreter.take_memory();
```

Comment:
This doesn't look like a good flow. The handler should allow overriding logic, so the best option would be for `interpreter.run` to be inside handler logic.

For this to be possible we should input instruction_tables directly to the handler call. I can make those changes if you want?

Reply:
I was thinking to *optionally* override, with fallback to default execution.

Reply:
It is a good alternative that you made, but in the future, i am building up to introduce generic call/create/eofcreate/auth (whatever) in that sense it should be called every time.